src/System.Management.Automation/engine/parser/ast.cs

```diff
@@ -7098,6 +7098,92 @@ internal virtual bool ShouldPreserveOutputInCaseOfException()
         }
     }
 
+    /// <summary>
+    /// The ast representing a ternary expression, e.g. <c>$a ? 1 : 2</c>.
+    /// </summary>
+    public class TernaryExpressionAst : ExpressionAst
+    {
+        /// <summary>
+        /// Construct a binary expression.
+        /// </summary>
+        /// <param name="extent">The extent of the expression.</param>
+        /// <param name="condition">The condition operand.</param>
+        /// <param name="ifTrue">The if clause.</param>
+        /// <param name="ifFalse">The else clause.</param>
+        public TernaryExpressionAst(IScriptExtent extent, ExpressionAst condition, ExpressionAst ifTrue, ExpressionAst ifFalse)
+            : base(extent)
+        {
+            if (condition == null || ifTrue == null || ifFalse == null)
+            {
+                throw PSTraceSource.NewArgumentNullException(condition == null ? nameof(condition) : ifTrue == null ? nameof(ifTrue) : nameof(ifFalse));
+            }
+
+            Condition = condition;
+            SetParent(condition);
+
+            IfTrue = ifTrue;
+            SetParent(ifTrue);
+
+            IfFalse = ifFalse;
+            SetParent(ifFalse);
+        }
+
+        /// <summary>
+        /// The ast for the condition of the ternary expression. The property is never null.
+        /// </summary>
+        public ExpressionAst Condition { get; }
+
+        /// <summary>
+        /// The ast for the if-operand of the ternary expression. The property is never null.
+        /// </summary>
+        public ExpressionAst IfTrue { get; }
+
+        /// <summary>
+        /// The ast for the else-operand of the ternary expression. The property is never null.
+        /// </summary>
+        public ExpressionAst IfFalse { get; }
+
+        /// <summary>
+        /// Copy the TernaryExpressionAst instance.
+        /// </summary>
+        public override Ast Copy()
+        {
+            var newCondition = CopyElement(this.Condition);
+            var newIfTrue = CopyElement(this.IfTrue);
+            var newIfFalse = CopyElement(this.IfFalse);
+            return new TernaryExpressionAst(this.Extent, newCondition, newIfTrue, newIfFalse);
+        }
+
+        #region Visitors
+
+        internal override object Accept(ICustomAstVisitor visitor)
+        {
+            var visitor2 = visitor as ICustomAstVisitor2;
+            return visitor2 != null ? visitor2.VisitTernaryExpression(this) : null;
+        }
+
+        internal override AstVisitAction InternalVisit(AstVisitor visitor)
+        {
+            var action = AstVisitAction.Continue;
+            if (visitor is AstVisitor2 visitor2)
+            {
+                action = visitor2.VisitTernaryExpression(this);
+                if (action == AstVisitAction.SkipChildren)
```

Comment:
Personal preference of course, but I would strongly support braces here

Reply:
Will add the braces.